File: 0078-subsets/0078-subsets.py
def subset(ind,n,arr,temp,res):
    if(ind>=n):
        res.append(temp[:])
        return
    
    temp.append(arr[ind])
    subset(ind+1,n,arr,temp,res)

    temp.pop()
    subset(ind+1,n,arr,temp,res)

class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        n = len(nums)
        temp =[]
        res = []
        subset(0,n,nums,temp,res)
        return res
        

File: 0078-subsets/0078-subsets.cpp
void printseqn(vector<int>& nums,int n, vector<vector<int>>& seqn,int index,vector<int>& vals){
    if(index == n){
        seqn.push_back(vals);
        return;
    }
    vals.push_back(nums[index]);
    printseqn(nums,n,seqn,index+1,vals);
    vals.pop_back();

    printseqn(nums,n,seqn,index+1,vals);
    return;
}

class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> seqn;
        vector<int> vals;
        int n = nums.size();
        printseqn(nums,n,seqn,0,vals);
        return seqn;
    }
};